### PR TITLE
feat(clerk-expo): Remove headers causing CORS errors

### PR DESCRIPTION
**Pull request by thiskevinwang**
## Description

This removes two headers, added in #2528, that are resulting in CORS errors being returned from the Clerk Frontend API.

Related:
- https://github.com/clerk/javascript/pull/2528#issuecomment-1962740121
- https://github.com/clerk/javascript/issues/2266

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/expo/src/singleton.ts [hunk 1]
```diff
@@ -43,13 +41,6 @@ export function buildClerk({ key, tokenCache }: BuildClerkOptions): HeadlessBrow
 
       const jwt = await getToken(KEY);
       (requestInit.headers as Headers).set('authorization', jwt || '');
-
-      // Send some non-identifying headers that are useful for logging
-      (requestInit.headers as Headers).set('x-expo-execution-environment', Constants.executionEnvironment);
-      (requestInit.headers as Headers).set(
-        'x-expo-native-application-version',
-        Application.nativeApplicationVersion || 'NULL',
-      );
```
Comment (panteliselef) [inline packages/expo/src/singleton.ts:52]: @thiskevinwang If having this kind of information was useful, and don't wanna completely removed it, couldn't we just do this ?

```ts
if(!isValidBrowser()) {
      // Send some non-identifying headers that are useful for logging
      (requestInit.headers as Headers).set('x-expo-execution-environment', Constants.executionEnvironment);
      (requestInit.headers as Headers).set(
        'x-expo-native-application-version',
        Application.nativeApplicationVersion || 'NULL',
      );
}
```

of use `import { Platform } from 'react-native';` 

```ts
if (Platform.OS !== 'web') {
      // Send some non-identifying headers that are useful for logging
      (requestInit.headers as Headers).set('x-expo-execution-environment', Constants.executionEnvironment);
      (requestInit.headers as Headers).set(
        'x-expo-native-application-version',
        Application.nativeApplicationVersion || 'NULL',
      );
}
```

Comment (thiskevinwang) [reply packages/expo/src/singleton.ts:52]: @panteliselef Oh, smart thinking! But also...

> If having this kind of information was useful

...Turns out, it was not particularly useful so I'd say we just remove this!

**Review by panteliselef — APPROVED** [review]
lgtm

**Review by anagstef — APPROVED** [review]

File: packages/expo/package.json
Comment (dimkl) [inline packages/expo/package.json:1]: @thiskevinwang You should also remove those dependencies from the `devDependencies`.

**Review by dimkl — APPROVED** [review]

File: packages/expo/package.json
Comment (thiskevinwang) [reply packages/expo/package.json:1]: Done in https://github.com/clerk/javascript/pull/3326/commits/05e82b497e9a8e98043ea64acb16da8a7892abf0

File: .changeset/sweet-beds-pump.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@clerk/clerk-expo': minor
+---
+
+Remove headers that were added in #2528
+
+Remove `expo-application` & `expo-constants` peer deps
```
[the remark this answers is not in the document]
Comment (thiskevinwang) [reply .changeset/sweet-beds-pump.md]: Updated in https://github.com/clerk/javascript/pull/3326/commits/05e82b497e9a8e98043ea64acb16da8a7892abf0

**Pull request merged**